hushline/premium.py

```diff
@@ -0,0 +1,609 @@
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        return render_template(
+            "premium-select-tier.html", user=user, business_price=get_business_price_string()
+        )
+
+    @bp.route("/select-tier/free", methods=["POST"])
+    @authentication_required
+    def select_free() -> Response | str:
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        if user.tier_id is None:
+            user.tier_id = FREE_TIER
+            db.session.add(user)
+            db.session.commit()
+
+        return redirect(url_for("inbox"))
+
+    @bp.route("/waiting", methods=["GET"])
+    @authentication_required
+    def waiting() -> Response | str:
+        return render_template("premium-waiting.html")
+
+    @bp.route("/upgrade", methods=["GET", "POST"])
+    @authentication_required
+    def upgrade() -> Response | str:
+        if request.method == "GET":
+            return redirect(url_for("premium.index"))
+
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        # If the user is already on the business tier
+        if user.tier_id == BUSINESS_TIER:
+            flash("👍 You're already upgraded.")
+            return redirect(url_for("premium.index"))
+
+        # Select the business tier
+        business_tier = db.session.get(Tier, BUSINESS_TIER)
+        if not business_tier:
+            current_app.logger.error("Could not find business tier")
+            flash("⚠️ Something went wrong!")
+            return redirect(url_for("premium.index"))
+        if not business_tier.stripe_price_id:
+            current_app.logger.error("Business tier does not have a price ID")
+            flash("⚠️ Something went wrong!")
+            return redirect(url_for("premium.index"))
+
+        # Make sure the user has a Stripe customer
+        try:
+            create_customer(user)
+        except stripe._error.StripeError as e:
+            current_app.logger.error(f"Failed to create Stripe customer: {e}")
+            flash("⚠️ Something went wrong!")
+            return redirect(url_for("premium.index"))
+
+        # Create a Stripe Checkout session
+        try:
+            checkout_session = stripe.checkout.Session.create(
+                client_reference_id=str(user.id),
+                customer=user.stripe_customer_id,
+                line_items=[{"price": business_tier.stripe_price_id, "quantity": 1}],
+                mode="subscription",
+                success_url=url_for("premium.waiting", _external=True),
+                automatic_tax={"enabled": True},
+                customer_update={"address": "auto"},
+            )
+        except stripe._error.StripeError as e:
```

Comment:
Might want `exc_info=True` in the logger to get more details instead of `{e}`

Reply:
Fixed in 977935a15167450c085c93340cb31e0ac6efca01